gmachine_opcodes_test.go

```diff
@@ -139,3 +139,52 @@ func TestSETA(t *testing.T) {
 		t.Errorf("want initial P value %d, got %d", wantP, g.P)
 	}
 }
+
+func TestCMP(t *testing.T) {
+	t.Parallel()
+	g := gmachine.New()
+	g.Memory[0] = gmachine.CMP
+	g.Memory[1] = 5
+	g.Run()
+	var wantT gmachine.Word = 0
+	if wantT != g.T {
+		t.Errorf("want initial T value %d, got %d", wantT, g.T)
+	}
+	var wantP gmachine.Word = 3
+	if wantP != g.P {
+		t.Errorf("want initial P value %d, got %d", wantP, g.P)
+	}
+}
+
+func TestCMPSetT(t *testing.T) {
+	t.Parallel()
+	g := gmachine.New()
+	g.Memory[0] = gmachine.INCA
```

Comment:
It would make sense to use your assembler for this, now that you've written it, wouldn't it? Since most tests run G-code programs, it's easier to supply them as a string, or slice of strings.